src/refmvs.rs

```diff
@@ -1582,40 +1578,39 @@ pub(crate) fn rav1d_refmvs_init_frame(
     n_tile_threads: u32,
     n_frame_threads: u32,
 ) -> Rav1dResult {
-    rf.sbsz = 16 << seq_hdr.sb128;
-    rf.iw8 = frm_hdr.size.width[0] + 7 >> 3;
-    rf.ih8 = frm_hdr.size.height + 7 >> 3;
-    rf.iw4 = rf.iw8 << 1;
-    rf.ih4 = rf.ih8 << 1;
-
-    let r_stride = ((frm_hdr.size.width[0] + 127 & !127) >> 2) as u32;
+    let rp_stride = ((frm_hdr.size.width[0] + 127 & !127) >> 3) as u32;
     let n_tile_rows = if n_tile_threads > 1 {
         frm_hdr.tiling.rows as u32
     } else {
         1
     };
-    let uses_2pass = (n_frame_threads > 1) as usize;
-    // `mem::size_of::<refmvs_block>() == 12`,
-    // but it's accessed using 16-byte loads in asm,
-    // so add `R_PAD` elements to avoid buffer overreads.
-    // TODO fallible allocation
-    rf.r.resize(
-        35 * r_stride as usize * n_tile_rows as usize * (1 + uses_2pass) + R_PAD,
-        FromZeroes::new_zeroed(),
-    );
-    rf.r_stride = r_stride;
+    let n_blocks = (rp_stride * n_tile_rows) as i32;
 
-    let rp_stride = r_stride >> 1;
-    // TODO fallible allocation
-    rf.rp_proj.resize(
-        16 * rp_stride as usize * n_tile_rows as usize,
-        Default::default(),
-    );
+    rf.sbsz = 16 << seq_hdr.sb128;
+    rf.iw8 = frm_hdr.size.width[0] + 7 >> 3;
+    rf.ih8 = frm_hdr.size.height + 7 >> 3;
+    rf.iw4 = rf.iw8 << 1;
+    rf.ih4 = rf.ih8 << 1;
     rf.rp_stride = rp_stride;
-
-    rf.n_tile_rows = n_tile_rows;
     rf.n_tile_threads = n_tile_threads;
     rf.n_frame_threads = n_frame_threads;
+
+    if n_blocks != rf.n_blocks {
+        let uses_2pass = (n_frame_threads > 1) as usize;
+        // `mem::size_of::<refmvs_block>() == 12`,
+        // but it's accessed using 16-byte loads in asm,
+        // so add `R_PAD` elements to avoid buffer overreads.
+        // TODO fallible allocation
+        rf.r.resize(
+            35 * 2 * n_blocks as usize * (1 + uses_2pass) + R_PAD,
+            FromZeroes::new_zeroed(),
+        );
+        // TODO fallible allocation
+        rf.rp_proj
```

Comment:
`rp_proj` is now a pointer to the latter section of `r`, so we're gonna need to do that in Rust somehow, too.

Reply:
Oh I see you said in the PR description we'll leave them separate for now.  Could you leave that as a TODO comment on the `rp_proj` field then?  It might be possible to do with `zerocopy` stuff, but we'll have to see if it's worth it.